classes/app.rb

```diff
@@ -1,35 +1,35 @@
 require_relative 'student'
 require_relative 'teacher'
 require_relative 'book'
+require_relative 'preserve_data'
 
 class App
   def initialize
-    @people = []
-    @books = []
-    @rentals = []
-  end
-
-  def ask_input(input)
-    puts input
+    @people = read_file('./data/people.json')
+    @books = read_file('./data/books.json')
+    @rentals = read_file('./data/rentals.json')
   end
 
   # option 1 - List all books
   def list_books
     puts 'No book yet!' if @books.empty?
-    @books.each { |book| puts "Title: \"#{book.title}\", Author: #{book.author}" }
+    @books.each { |book| puts "Title: \"#{book['title']}\", Author: \"#{book['author']}\" Rentals: #{book['rentals']}" }
   end
 
   # option 2 - List all people
   def list_people
     puts 'We do not have people yet' if @people.empty?
-    @people.each { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+    @people.each do |person|
+      puts "Name: #{person['name']}, ID: #{person['id']}, Age: #{person['age']}, Rentals: #{person['rentals']}"
+    end
   end
```

Comment:
- [ ] Hey guys, really great job so far! But It seems there's a bug in the above 2 functions for listing books and people. Immediately after creating a person or a book If the user tries to list all people or books the below error is shown but after restarting the app It retrieves the data fine. After debugging your code for a few minutes I found that the data from the files are only retrieved when the app first starts in the constructor. So this means any updates to the file after startup won't be properly retrieved to avoid this I'd suggest reading from the file inside the functions instead of the constructor as shown in the code snippet below. 😉
- These are the errors shown:
  ![image](https://github.com/kkmanuu/OOP-school-library/assets/104968236/4b8b5e9f-f7eb-4018-8899-00d1c9932320)
  ![image](https://github.com/kkmanuu/OOP-school-library/assets/104968236/d2f2a74f-0080-43c3-8680-675dcda457a7)
- Code snippets:
   ```rb
     def list_books
         @books = read_file('./data/books.json')
         puts 'No book yet!' if @books.empty?
         @books.each { |book| puts "Title: \"#{book['title']}\", Author: \"#{book['author']}\" Rentals: #{book['rentals']}" }
     end
      def list_people
         @people = read_file('./data/people.json')
         puts 'We do not have people yet' if @people.empty?
         @people.each do |person|
           puts "Name: #{person['name']}, ID: #{person['id']}, Age: #{person['age']}, Rentals: #{person['rentals']}"
         end
       end
   ```

Reply:
@AbelG101 Done making the changes